apps/dot-cdn/src/app/dotcdn.store.ts

```diff
@@ -0,0 +1,176 @@
+}
+
+export const enum Loader {
+    CHART = 'CHART',
+    PURGE_URLS = 'PURGE_URLS',
+    PURGE_PULL_ZONE = 'PURGE_PULL_ZONE'
+}
+
+@Injectable()
+export class DotCDNStore extends ComponentStore<DotCDNState> {
+    constructor(private readonly dotCdnService: DotCDNService) {
+        super({
+            chartData: {
+                labels: [],
+                datasets: []
+            },
+            statsData: [],
+            isChartLoading: LoadingState.IDLE,
+            isPurgeUrlsLoading: LoadingState.IDLE,
+            isPurgeZoneLoading: LoadingState.IDLE
+        });
+    }
+    readonly vm$ = this.select(
+        ({ chartData, statsData, isChartLoading, isPurgeUrlsLoading, isPurgeZoneLoading }) => ({
+            chartData,
+            statsData,
+            isChartLoading: isChartLoading === LoadingState.LOADING,
+            isPurgeUrlsLoading: isPurgeUrlsLoading === LoadingState.LOADING,
+            isPurgeZoneLoading: isPurgeZoneLoading === LoadingState.LOADING
+        })
+    );
+
+    /**
+     *
+     *  Adds chart data
+     * @memberof DotCDNStore
+     */
+    readonly addChartData = this.updater((state, chartData: ChartData) => {
+        return {
+            ...state,
+            chartData
+        };
+    });
+
+    /**
+     *
+     *  Adds stats data
+     * @memberof DotCDNStore
+     */
+    readonly addStatsData = this.updater((state, statsData: DotChartStats[]) => {
+        return {
+            ...state,
+            statsData
+        };
+    });
+
+    /**
+     *  Dispatches a loading state
+     *
+     * @memberof DotCDNStore
+     */
+    readonly dispatchLoading = this.updater(
+        (state, action: { loadingState: string; loader: string }) => {
+            console.log(action.loader, action.loadingState);
+            switch (action.loader) {
+                case Loader.CHART:
+                    return {
+                        ...state,
+                        isChartLoading: action.loadingState
+                    };
+                case Loader.PURGE_URLS:
+                    return {
+                        ...state,
+                        isPurgeUrlsLoading: action.loadingState
+                    };
+                case Loader.PURGE_PULL_ZONE:
+                    return {
+                        ...state,
+                        isPurgeZoneLoading: action.loadingState
+                    };
+            }
+        }
+    );
+
+    /**
+     *  Handles the chart data fetching
+     *
+     * @memberof DotCDNStore
+     */
+    readonly getChartStats = this.effect((period$: Observable<string>) => {
+        return period$.pipe(
+            mergeMap((period) => {
+                // Dispatch the loading state
+                this.dispatchLoading({
+                    loadingState: LoadingState.LOADING,
+                    loader: Loader.CHART
+                });
+                return this.dotCdnService.requestStats(period).pipe(
+                    tap({
```

Comment:
Their doc is bad but they created a operator for this: https://ngrx.io/guide/component-store/effect#tapresponse